From 9e2de8515be15e2b61ae518ce592e5b446ef2d48 Mon Sep 17 00:00:00 2001
From: Sander Vanheule <sander@svanheule.net>
Date: Sun, 25 Jun 2023 20:55:00 +0200
Subject: [PATCH] tplink-safeloader: add EAP610 v3 and EAP613 v1

The TP-Link EAP610 v3 is a MT7621 version of the EAP610 v1/v2, which use
a QCA chipset. The TP-Link EAP613 v1 appears to be rebrand of the
former.

The original 'os-image' and 'file-system' partition are merged into a
single 'firmware' partition.

Signed-off-by: Sander Vanheule <sander@svanheule.net>
---
 src/tplink-safeloader.c | 33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/src/tplink-safeloader.c b/src/tplink-safeloader.c
index 7a15717..53a42f7 100644
--- a/src/tplink-safeloader.c
+++ b/src/tplink-safeloader.c
@@ -2121,6 +2121,39 @@ static struct device_info boards[] = {
 		.last_sysupgrade_partition = "file-system"
 	},
 
+	/** Firmware layout for the EAP610 v3/EAP613 v1 */
+	{
+		.id = "EAP610-V3",
+		.soft_ver = SOFT_VER_DEFAULT,
+		.soft_ver_compat_level = 1,
+		.support_list =
+			"SupportList:\r\n"
+			"EAP610(TP-Link|UN|AX1800-D):3.0\r\n"
+			"EAP610(TP-Link|JP|AX1800-D):3.0\r\n"
+			"EAP610(TP-Link|EG|AX1800-D):3.0\r\n"
+			"EAP610(TP-Link|CA|AX1800-D):3.0\r\n"
+			"EAP613(TP-Link|UN|AX1800-D):1.0 JP\r\n",
+		.part_trail = PART_TRAIL_NONE,
+
+		.partitions = {
+			{"fs-uboot", 0x00000, 0x80000},
+			{"partition-table", 0x80000, 0x02000},
+			{"default-mac", 0x90000, 0x01000},
+			{"support-list", 0x91000, 0x00100},
+			{"product-info", 0x91100, 0x00400},
+			{"soft-version", 0x92000, 0x00100},
+			{"firmware", 0xa0000, 0xcf0000},
+			{"user-config", 0xd90000, 0x60000},
+			{"mutil-log", 0xf30000, 0x80000},
+			{"oops", 0xfb0000, 0x40000},
+			{"radio", 0xff0000, 0x10000},
+			{NULL, 0, 0}
+		},
+
+		.first_sysupgrade_partition = "os-image",
+		.last_sysupgrade_partition = "file-system"
+	},
+
 	/** Firmware layout for the EAP615-Wall v1 */
 	{
 		.id = "EAP615-WALL-V1",
